src/EventListener/Isotope/PostOrderStatusUpdate/AddMemberGroupsListener.php

```diff
@@ -22,19 +23,25 @@ public function __construct(
     ) {
     }
 
-    public function __invoke(Order $order, array $tokens): void
+    public function __invoke(Order $order, int $oldStatus, OrderStatus $newStatus): void
     {
-        if (!($user = $this->security->getUser()) instanceof FrontendUser) {
+
+        if (!$newStatus->isPaid()) {
             return;
         }
 
+        if (($memberId = $order->member) <= 0) {
+            return;
+        }
+
+        $member = MemberModel::findByPk($memberId);
```

Comment:
```suggestion
        if (!$member = MemberModel::findByPk($order->member)) {
            return;
        }

```

The member might have been deleted.